client/solve.go

```diff
@@ -90,7 +91,19 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
 		return nil, errors.New("invalid with def and cb")
 	}
 
-	syncedDirs, err := prepareSyncedDirs(def, opt.LocalDirs)
+	sourceFS := make(map[string]fsutil.FS)
+	for k, fs := range opt.LocalFiles {
+		sourceFS[k] = fs
+	}
+	for k, dir := range opt.LocalDirs {
+		fs, err := fsutil.NewFS(dir)
+		if err != nil {
+			return nil, err
+		}
+		sourceFS[k] = fs
+	}
```

Comment:
I think it would be better to move this into a function. `mergeLocalX` with `X` being the name we decide for the solve opt name. So if it's local mounts, `mergeLocalMounts`, `createLocalMounts`, or maybe just `createSourceFS`?

Alternatively, since `LocalDirs` is deprecated, maybe we shouldn't implement this merging behavior? If `LocalFiles` is defined, use it exclusively. If it's not, fallback on `LocalDirs`.

Reply:
> I think it would be better to move this into a function. `mergeLocalX` with `X` being the name we decide for the solve opt name. So if it's local mounts, `mergeLocalMounts`, `createLocalMounts`, or maybe just `createSourceFS`?

Given this is such a short snippet and not used anywhere, I don't see the need to move it - I've annotated with a comment though.

> Alternatively, since `LocalDirs` is deprecated, maybe we shouldn't implement this merging behavior? If `LocalFiles` is defined, use it exclusively. If it's not, fallback on `LocalDirs`.

If we're aiming to have a release in which both are supported, then the most obvious behavior to me as a user is that setting both result in both getting set. The scenarios it makes sense to choose one over the other exclusively is at the gRPC API level, when any well-behaved client cannot set both. If we're aiming to not release anything where both are set (e.g. in a follow-up removing `opt.LocalDirs`), then it doesn't really matter the behavior atm.

Reply:
I generally prefer a function if it's possible. It reduces the number of live local variables, self-documents itself, and also reduces the complexity for reading the function. In isolation, this snippet may be simple. In the overall context, each small bit of reduced complexity helps someone understand the intention of the top level function. In this case, I think this behavior is inconsequential to understanding `solve` so it's helpful to just remove it.